windows-containers/productization.md

```diff
@@ -0,0 +1,187 @@
+that all Windows container images contains a Windows kernel and Red Hat has a
+policy to not ship 3rd party kernels for support reasons.
+
+### Implementation Details
+
+#### Windows Scale Up (WSU)
+
+The Windows Scale Up is an Ansible playbook that has the follow prerequisites:
+* Needs to run on a Linux system
+* Needs to be able to access the cluster where the Windows worker will be added
+* Needs to be able to access the Windows node
+
+The inputs that the WSU playbook requires are:
+* Windows node credentials
+* Kubelet download location
+* Worker ignition
+* WMCB download location
+
+The actions that the WSU will perform are:
+* Check if the container runtime is present on the Windows node
+* Download and copy the kubelet to the Windows node
+* Extract the worker ignition from the cluster and copy it to the Windows node
+* Download the WMCB, copy it to the Windows node and execute it
+
+#### Windows Machine Config Bootstrapper (WMCB)
+
+The Windows Machine Config Bootstrapper executable has the following
+prerequisites:
+* Only supports Windows x86-64 nodes (Windows Server 2019)
+* Can only run on Windows nodes
+
+The inputs that the WMCB requires are:
+* kubelet location on the local disk
+* Worker ignition location on the local disk
+
+The actions that the WMCB will perform are:
+* Install / upgrade and configure the kubelet
+* Parse the worker ignition and extract the bootstrap kubeconfig and the
+kubelet configuration
+  * We are not using the ignition file during the node booting stage
+* Launch the kubelet as a Windows service
+* Check if the kubelet is running
+* Exit
+
+### Risks and Mitigations
+
+The main risk with this proposal is the dependency on Microsoft to publish a
+downstream version of the kubelet. If this does not happen we would have to
+use the upstream version of the kubelet which will result in Red Hat being
+responsible for security and other fixes.
+
+The other risk is "OVN Plumbing for Hybrid Linux+Windows Clusters GA" not being
+delivered in time for integration. The mitigation for that would be to use 3rd
+party networking components but that might have support implications that would
+need to be resolved.
+
+## Design Details
+
+### Test Plan
+
+We plan to have all the repositories associated with this effort fully
+integrated with Prow CI and run e2e tests for every PR that is opened. These
+e2e tests will involve bringing up a cluster on all supported cloud providers,
+instantiating a Windows node and running workloads on it. We also plan to add
+blocking tests to the nightly runs.
+
+### Graduation Criteria
+
+This enhancement will start as GA
+
+### Upgrade / Downgrade Strategy
+
+We will support upgrades of the node components by publishing a new release
+of the WSU Ansible playbook. An older release of the playbook can be used to
+downgrade.
+
+### Version Skew Strategy
+
+We plan to maintain kubelet major version parity with the Linux counterpart.
```

Comment:
You'll have to have skew during upgrades across the boundary though.  Maybe say something about having CI coverage when we get to that point so you can ensure that at least some Windows kubelets will successfully handle lagging by one major version behind the rest of the cluster.  Or say that before a major bump folks might have to drain/wipe their Windows compute and re-attach them as fresh compute after the major bump?

Reply:
We are planning on advising the customer to exactly what is done in the BYO RHEL scenario i.e. drain the Windows compute nodes using the Ansible playbook. I will add a line about that.